src/common/ColourPalette.hpp

```diff
@@ -31,15 +33,53 @@ class ColourPalette {
         /** Converts a given palette value into packed RGB (format 0x00RRGGBB). */
         uInt32 getRGB(int val) const;
 
-    private:
+        /** returns a pointer to the palette array (256 elements) */
+        const uInt32 *getPalette();
+
+        /**
+          Loads all defined palettes with PAL color-loss data depending
+          on 'state'.
+          Sets the palette according to the given palette name.
+
+          @param type  The palette type = {standard, z26, user}
+          @param displayFormat The display format = { NTSC, PAL, SECAM }
+        */
+        void setPalette(const string& type,
+                        const string& displayFormat);
+
+        /**
+            Loads a user-defined palette file (from OSystem::paletteFile), filling the
+            appropriate user-defined palette arrays.
+        */
+        void loadUserPalette(const string& paletteFile);
+
+private:
+
+        /**
+         *  Calculates grayscale values for all palettes
+         */
+        void calculateGrayscaleValues();
+
+        uInt32 *m_palette;
+
+        bool myUserPaletteDefined;
+
+        // Table of RGB values for NTSC, PAL and SECAM
+        static uInt32 NTSCPalette[256];
```

Comment:
Sure, I will try to make the interface cleaner. Still... 
- loadUserPalette is used outside the class. It is necessary.
- m_palette needs to be a member. Each ColourPalette instance needs each own pointer to the buffer. I guess that at some point, exporting might need to create its own ColourBuffer, etc. this is unless you want to have a singleton global palette, which I am not sure if it makes sense.
- calculateGrayscaleValues(). That would work for standard and z26 but not for user palettes, which are instance related. Again, unless you want the palette object to be a singleton.
- NTSC/etc. I can move them out..